### PR TITLE
feat(rollup-plugin-index-html): enable inline module scripts

### PR DESCRIPTION
**Pull request by bennypowers**
Not sure if this is the best approach (fs.writeFileSync), but here's an ostensibly working implementation of inline importmapped modules for rollup-plugin-index-html for your consideration

### CODE REVIEW

**Comment by LarsDenBakker** [comment]
Looks like a good approach, but I want to think about this a bit and look at the use cases.

We need to keep the logic inside `rollup-plugin-index-html` to a minimum, and keep most of the logic in `building-utils`. It needs to work for the webpack plugin too, and es dev server (which is in a sperate PR but also uses these building-utils functions).

File: packages/rollup-plugin-index-html/src/process-entry-html.js [hunk 1]
```diff
@@ -16,15 +17,27 @@ function processEntryHtml(inputConfig) {
...
+    jsModules,
+    inlineScripts,
+    inlineImportMaps,
+    indexHTML: outputIndexHTML,
+  } = extractResources(indexHTMLString);
+  const input = jsModules.map(p => path.join(indexFolder, p));
+  if (inlineScripts.length) {
+    inlineScripts.forEach(script => {
+      const inlineScriptPath = path.join(indexFolder, `inline.${md5(script)}.js`);
+      fs.writeFileSync(inlineScriptPath, script);
```
Comment (LarsDenBakker) [inline packages/rollup-plugin-index-html/src/process-entry-html.js]: rollup will already output whatever is in the input, so we don't need to write it ourselves.

File: packages/rollup-plugin-index-html/src/process-entry-html.js [hunk 2]
```diff
@@ -16,15 +17,27 @@ function processEntryHtml(inputConfig) {
...
+  const {
+    jsModules,
+    inlineScripts,
+    inlineImportMaps,
+    indexHTML: outputIndexHTML,
+  } = extractResources(indexHTMLString);
+  const input = jsModules.map(p => path.join(indexFolder, p));
+  if (inlineScripts.length) {
+    inlineScripts.forEach(script => {
+      const inlineScriptPath = path.join(indexFolder, `inline.${md5(script)}.js`);
```
Comment (LarsDenBakker) [inline packages/rollup-plugin-index-html/src/process-entry-html.js]: rollup will hash the final script output, you're now hashing the unbuilt content. I think we can just use the script's array index as id. (`inline-${i}.js`)

**Comment by LarsDenBakker** [comment]
Can you add a test fixture of an inline script + non-inline script + legacy build? :)

File: packages/rollup-plugin-index-html/test/integration.test.js [hunk 3]
```diff
@@ -33,6 +33,12 @@ async function testSnapshot(name, configFilePath) {
     });
   }
 
+  if (name.includes('inline-entry')) {
```
Comment (LarsDenBakker) [inline packages/rollup-plugin-index-html/test/integration.test.js]: `testSnapshots` below already does the same I tink

File: packages/rollup-plugin-index-html/src/process-entry-html.js (hunk 1 above)
Comment (bennypowers) [reply packages/rollup-plugin-index-html/src/process-entry-html.js]: we need to write here because we're *creating* the input

Comment (LarsDenBakker) [reply packages/rollup-plugin-index-html/src/process-entry-html.js]: Ouch... I hope there is a cleaner way for thag

File: packages/rollup-plugin-index-html/src/process-entry-html.js (hunk 2 above)
Comment (bennypowers) [reply packages/rollup-plugin-index-html/src/process-entry-html.js]: fine by me

File: packages/rollup-plugin-index-html/test/integration.test.js (hunk 3 above)
Comment (bennypowers) [reply packages/rollup-plugin-index-html/test/integration.test.js]: this whole block was committed by mistake

File: packages/rollup-plugin-index-html/src/process-entry-html.js (hunk 1 above)
Comment (bennypowers) [reply packages/rollup-plugin-index-html/src/process-entry-html.js]: unless there's some way to pass script content as a string to input, but I think it only takes a list of pathnames

**Comment by bennypowers** [comment]
@LarsDenBakker in channel we talked about 
- can we assume the user really wants us to extract inline scripts to files? should we instead leave some comment in the dom with a hash of the script and pass that to the plugin, which will replace the inline content with the built version?
- doing more work in build-utils and less in the rollup-plugin, specifically, I imagine you're proposing writing inline modules as files in `@open-wc/building-utils/index-html/extract-resources.js` instead of in rollup?

File: packages/rollup-plugin-index-html/src/process-entry-html.js (hunk 1 above)
Comment (bennypowers) [reply packages/rollup-plugin-index-html/src/process-entry-html.js]: could be we could defer all that work and handle it all in generateBundle or writeBundle, which is probably The Right Way™️, but also more tightly couples the thing to rollup

**Comment by LarsDenBakker** [comment]
After thinking about it, I think we can transform inline module scripts by default and add an option to turn that behavior off. 

I'd prefer not to write any inline modules to files at all, so I guess that takes moving more logic to building-utils out of the equation... I'll check if there is a rollup way for it

**Comment by LarsDenBakker** [comment]
@bennypowers I've pushed changes to avoid writing to temporary file.

I think this is good to go now, can you rebase on master and make this into one commit?

**Pull request merged**